pkgs/servers/sql/postgresql/ext/pgddl.nix

```diff
@@ -0,0 +1,47 @@
+{
+  lib,
+  fetchFromGitHub,
+  postgresql,
+  buildPostgresqlExtension,
+  postgresqlTestExtension,
+  perl,
+}:
+buildPostgresqlExtension (finalAttrs: {
+  pname = "pgddl";
+  version = "0.29";
+
+  src = fetchFromGitHub {
+    owner = "lacanoid";
+    repo = "pgddl";
+    tag = finalAttrs.version;
+    hash = "sha256-W3G6TGtkj+zXXdGZZR0bmZhsLuFJvuGTlDoo8kL8sf0=";
+  };
+
+  buildInputs = [ perl ];
```

Comment:
I think this belongs to `nativeBuildInputs`: short version, dependencies that get executed during the nix-build belong to `nativeBUildInputs`, dependencies that are used at runtime belong to `buildInputs`. This distinction is relevant for cross builds[1].

Given that the perl scripts are apparently executed during the build (and you correctly used `patchShebangs --build`), this should be changed here. Also, `postgresql` belongs to `nativeBuildInputs`: `buildPostgresqlExtension` only sets it into `buildInputs` because most extensions link against it, but here we run `pg_config` on build-time.

You can test those things with `strictDeps = true;` for correctness btw. Might be reasonable to also set it here unless I'm missing something.

Reply:
Updated to include `strictDeps = true` and moved to `nativeBuildInputs`

Reply:
Yeah, that'd be out of scope here, thanks!

Merging anyways, I don't think it's worth blocking this on the refactor, we can rebase later anyways.